File: app/src/main/java/com/example/thomasraybould/nycschools/view/school_list_activity/SchoolListPresenter.java
package com.example.thomasraybould.nycschools.view.school_list_activity;

import com.example.thomasraybould.nycschools.adapters.school_list_adapter.SchoolListItemUiModel;
import com.example.thomasraybould.nycschools.view.base.Presenter;

public interface SchoolListPresenter extends Presenter<SchoolListView>{

    void onSchoolListItemSelected(SchoolListItemUiModel schoolListItemUiModel);
}

File: app/src/main/java/com/example/thomasraybould/nycschools/view/school_list_activity/SchoolListPresenter.kt
package com.example.thomasraybould.nycschools.view.school_list_activity

import androidx.lifecycle.LiveData
import com.example.thomasraybould.nycschools.adapters.school_list_adapter.SchoolListItemUiModel
import com.example.thomasraybould.nycschools.view.base.Presenter

interface SchoolListPresenter : Presenter<SchoolListView> {
    fun getSchoolList(): LiveData<SchoolListUiModel>
    fun onSchoolListItemSelected(schoolListItemUiModel: SchoolListItemUiModel)
}
